frontend/src/metabase-lib/v1/expressions/recursive-parser.js

```diff
@@ -357,7 +358,7 @@ export const adjustMultiArgOptions = tree =>
 export const adjustBooleans = tree =>
   modify(tree, node => {
     if (Array.isArray(node)) {
-      if (node?.[0] === "case") {
+      if (isCaseOrIfOperator(node[0])) {
```

Comment:
Not sure if it was an intentional change but the new code does not accommodate for cases where `node` is nullish.